controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/RoleMembership.java

```diff
@@ -52,4 +58,59 @@ public String toString() {
         RoleMembership membership(Principal user);
     }
 
+    public interface Builder {
+
+        BuilderWithRole add(Role role);
+
+        RoleMembership build();
+
+    }
+
+    public static class BuilderWithRole implements Builder {
```

Comment:
How about `RoleBuilder`?

Reply:
Perhaps. It's not exactly that, either, since you can still `build()` it and get a `RoleMembership`. Anyway, as you see from the usage, it shouldn't be necessary to declare variables of this kind anyway :)